### PR TITLE
Implementations should support zstd

### PR DESCRIPTION
**Pull request by sudo-bmitch**
Partial solution to #803.

### CODE REVIEW

**Comment by sudo-bmitch** [comment]
CI will be fixed by #1199

File: manifest.md [hunk 1]
```diff
@@ -85,10 +85,15 @@ Unlike the [image index](image-index.md), which contains information about a set
...
 
     Manifests concerned with portability SHOULD use one of the above media types.
-    Implementations storing or copying image manifests MUST NOT error on encountering a `mediaType` that is unknown to the implementation.
-
     Entries in this field will frequently use the `+gzip` types.
 
+    Implementations SHOULD also support the following media types:
+
+    - [`application/vnd.oci.image.layer.v1.tar+zstd`](layer.md#zstd-media-types)
+    - [`application/vnd.oci.image.layer.nondistributable.v1.tar+zstd`](layer.md#zstd-media-types)
```
[the remark this answers is not in the document]
Comment (sudo-bmitch) [reply manifest.md:93]: They are deprecated in `media-types.md`. I agree with the question of why we didn't deprecate them in other parts of the spec.

[the remark this answers is not in the document]
Comment (tianon) [reply manifest.md:93]: So perhaps we start here by not listing this zstd variant? :innocent:

[the remark this answers is not in the document]
Comment (sudo-bmitch) [reply manifest.md:93]: My OCD would bug me leaving the file in a half complete state. I'll make a separate PR finishing the deprecation, and then rebase this PR on that.

[the remark this answers is not in the document]
Comment (sudo-bmitch) [reply manifest.md]: I've rebased on top of #1200. I'm happy to discuss over there if we should drop the MUST requirement for those media types down to a MAY over there, and if we do that, I'd do the same for the nondistributable zstd layer.

File: manifest.md [hunk 2]
```diff
@@ -81,14 +81,19 @@ Unlike the [image index](image-index.md), which contains information about a set
...
 
     Manifests concerned with portability SHOULD use one of the above media types.
-    Implementations storing or copying image manifests MUST NOT error on encountering a `mediaType` that is unknown to the implementation.
-
     Entries in this field will frequently use the `+gzip` types.
 
+    Implementations SHOULD also support the following media types:
+
+    - [`application/vnd.oci.image.layer.v1.tar+zstd`](layer.md#zstd-media-types)
+    - [`application/vnd.oci.image.layer.nondistributable.v1.tar+zstd`](layer.md#zstd-media-types) ([deprecation notice](./layer.md#non-distributable-layers))
```
[the remark this answers is not in the document]
Comment (tianon) [reply manifest.md:93]: It reads to me like "hey, new version of the spec says you SHOULD support these media types, but one of them is already deprecated at release time!"

[the remark this answers is not in the document]
Comment (sudo-bmitch) [reply manifest.md]: Rebased and deleted that.

**Review by tianon — APPROVED** [review]

**Review by sajayantony — APPROVED** [review]

**Pull request merged**